openedx_tagging/core/tagging/models.py

```diff
@@ -136,20 +140,109 @@ class Taxonomy(models.Model):
             "Indicates that tags in this taxonomy need not be predefined; authors may enter their own tag values."
         ),
     )
+    system_defined = models.BooleanField(
+        default=False,
+        editable=False,
+        help_text=_(
+            "Indicates that tags and metadata for this taxonomy are maintained by the system;"
+            " taxonomy admins will not be permitted to modify them.",
+        ),
+    )
+    visible_to_authors = models.BooleanField(
+        default=True,
+        editable=False,
+        help_text=_(
+            "Indicates whether this taxonomy should be visible to object authors."
+        ),
+    )
+    _taxonomy_class = models.CharField(
+        null=True,
+        max_length=255,
+        help_text=_(
+            "Taxonomy subclass used to instantiate this instance."
+            "Must be a fully-qualified module and class name.",
```

Comment:
What will the behavior be if the taxonomy class is no longer found? It would be good to state that explicitly here.

Plugins will come and go, so we need to be sure that falling back to the "generic" behavior will always at least preserve existing tags, but perhaps not allow new tags to be added. I believe you already have implemented this and have a test for it; I just think it's good to explain here as well.

Reply:
Good point, done: https://github.com/openedx/openedx-learning/pull/62/commits/1094d24c5f99a375144b13171c88efec27a9244b#diff-204875f41064da3abfd4648611ad41c6f21abe09277263410e3daafbd343f2a3R162-R163